pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -313,6 +324,30 @@ func (dsw *desiredStateOfWorld) VolumeExists(
 	return false
 }
 
+// DetermineVolumeAction determines action to be taken for pod volume
+// if the volume should be added or removed from desired state of world.
+// return values:
+//    true  -- if volume needs to be attached
+//    false -- if volume needs to be detached
+func (dsw *desiredStateOfWorld) DetermineVolumeAction(pod *v1.Pod, defaultValue bool) bool {
```

Comment:
yeah I was debating between moving this to util vs keeping in dsw. The problem is - dswp in this case holds `managedNode` and is a unexported member of `cache` package and hence can't be accessed directly. 

In this case dws along with pod holds the truth about volume action (attach or detach) needed on the pod.  I am okay moving this to util, but we will have to export the `managedNode` type from `cache` package or something similar.

Having said that - what are your thoughts on this?